services/src/main/java/org/keycloak/services/managers/DefaultBruteForceProtector.java

```diff
@@ -214,7 +227,15 @@ public boolean isTemporarilyDisabled(KeycloakSession session, RealmModel realm,
 
     @Override
     public boolean isPermanentlyLockedOut(KeycloakSession session, RealmModel realm, UserModel user) {
-        return !user.isEnabled() && DISABLED_BY_PERMANENT_LOCKOUT.equals(user.getFirstAttribute(DISABLED_REASON));
+        if (!user.isEnabled() && DISABLED_BY_PERMANENT_LOCKOUT.equals(user.getFirstAttribute(DISABLED_REASON))) {
+            return true;
+        }
+
+        if (!realm.isPermanentLockout()) return false;
+
+        // recheck failures just in case we are in a race
```

Comment:
Did you mean to include this in the PR?

Reply:
I included it here although it's not related. I detected one test was still failing from time to time because of this and I added the change here. We can remove it and send it in another PR/issue if needed.

Reply:
I have rebased to the last main. There were conflicts.